autoswitch_virtualenv.plugin.zsh

```diff
@@ -279,6 +279,23 @@ function install_requirements() {
         fi
     fi
 
+    if [[ -f "$PWD/pyproject.toml" ]]; then
+        printf "Found a ${PURPLE}pyproject.toml${NORMAL} file. Install dependencies? [y/N]: "
+        read ans
+
+        if [[ "$ans" = "y" || "$ans" = "Y" ]]; then
+            if [[ "$AUTOSWITCH_PIPINSTALL" = "FULL" ]]
+            then
+                pip install .
+            else
+                printf "Found a pyproject.toml file, but editable mode currently requires a setup.py based build.\n"
+                printf "Set AUTOSWITCH_PIPINSTALL to FULL to disable editable mode.\n"
+                printf "\n"
+                return
```

Comment:
this message works but exists too early. It should exit *before* the `.venv` file is created to prevent the user's system ending up in a weird state

Reply:
When `AUTOSWITCH_PIPINSTALL` is not set to `FULL`, and when encountering a pyproject.toml file, we have three options when trying to install requirements:
1) Ignore the `AUTOSWITCH_PIPINSTALL` setting, and simply run `pip install .`. Upside: it will work. Downside: it is confusing for users (which could be solved with a message, but still).
2) Don't run any `pip install`, but just display the message indicating users should set `AUTOSWITCH_PIPINSTALL` to `FULL`. It will create a `.venv`, but just not install dependencies.
3) Don't create `.venv`, don't (try to) install dependencies, only display a message.

Between 2 and 3, I think 2 is easier to implement and with a decent message for users, it should also be understandable to them. Besides, they still have the chance of running `pip install .` by themselves in the new virtual env.

Reply:
I think 1) is actually my preferred option in this case. While it is slightly confusing, it seems like if `pyproject.toml` does not support the alternative option then we should just use what works without interrupting the user.

Reply:
I guess you are right. The name of the `AUTOSWITCH_PIPINSTALL` variable made it sound like something that has influence on various decisions, but it really is only there to distinguish between `pip install` in editable mode or not for setup.py projects. The name of the variable is a bit confusing perhaps.

Anyway, I'll make the changes as suggested (option 1).

Reply:
Yeah I agree, the configuration name is a bit of a misnomer. Might make sense to try migrate to a different name.